route-rs-runtime/src/link/tee_link.rs

```diff
@@ -1,50 +1,103 @@
 use crate::link::task_park::*;
-use crate::link::{AsyncEgressor, PacketStream};
+use crate::link::{AsyncEgressor, Link, LinkBuilder, PacketStream};
 use crossbeam::atomic::AtomicCell;
 use crossbeam::crossbeam_channel;
 use crossbeam::crossbeam_channel::{Receiver, Sender};
 use futures::{Async, Future, Poll, Stream};
 use std::sync::Arc;
 
-pub struct TeeLink<P: Sized + Clone> {
-    pub ingressor: TeeIngressor<P>,
-    pub egressors: Vec<AsyncEgressor<P>>,
+#[derive(Default)]
+pub struct TeeLinkBuilder<Packet: Sized + Clone + Send> {
+    in_stream: Option<PacketStream<Packet>>,
+    queue_capacity: usize,
+    branches: Option<usize>,
 }
 
-impl<P: Sized + Clone> TeeLink<P> {
-    pub fn new(input_stream: PacketStream<P>, queue_capacity: usize, branches: usize) -> Self {
+impl<Packet: Sized + Clone + Send> TeeLinkBuilder<Packet> {
```

Comment:
We could also define an `ingressor` method for `TeeLink` that works similarly to other `Link`s that take just one ingressor.

Reply:
I feel like it is kludgy to have special methods in general for certain links,  would like to discuss with the group in person, see what everyone thinks.

Reply:
Doine